core/server/logging/PrettyStream.js

```diff
@@ -68,84 +68,77 @@ PrettyStream.prototype.write = function write(data) {
+        // can be extended to define from outside, but not important
+        try {
+            output += format('%s %s [%s] "%s %s" %s %s\n',
+                logLevel,
+                data.req.ip,
+                time,
+                data.req.method.toUpperCase(),
+                data.req.originalUrl,
+                data.res.statusCode,
+                data.res._headers['x-response-time']
+            );
+        } catch (err) {
+            output += format('[%s] %s\n',
+                time,
+                logLevel
+            );
+        }
 
-            if (key === 'level') {
-                bodyPretty += colorize('underline', key + ':' + value) + '\n';
-            }
-            else if (key === 'message') {
-                bodyPretty += colorize('red', value) + '\n';
-            }
-            else if (key === 'context') {
-                bodyPretty += colorize('white', value) + '\n';
-            }
-            else if (key === 'help') {
-                bodyPretty += colorize('yellow', value) + '\n';
+        _.each(_.omit(data, ['time', 'level', 'name', 'hostname', 'pid', 'v', 'msg']), function (value, key) {
+            bodyPretty += '\n' + colorize('yellow', key.toUpperCase()) + '\n';
+
+            // we always output errors for now
+            if (_.isObject(value) && value.message && value.stack) {
+                var error = colorize('red', value.message) + '\n';
+
+                if (value.context) {
+                    error += colorize('white', value.context) + '\n';
+                }
+
+                if (value.help) {
+                    error += colorize('yellow', value.help) + '\n';
+                }
+
+                if (value.stack && !value.hideStack) {
+                    error += colorize('white', value.stack) + '\n';
+                }
+
+                output += format('%s\n', colorize('red', error));
             }
-            else if (key === 'stack' && !data.err['hideStack']) {
-                bodyPretty += colorize('white', value) + '\n';
+            else if (_.isObject(value)) {
+                var sanitized = {};
+
+                _.each(value, function (innerValue, innerKey) {
+                    if (!_.isEmpty(innerValue)) {
+                        sanitized[innerKey] = innerValue;
+                    }
+                });
+
+                bodyPretty += prettyjson.render(sanitized, {}) + '\n';
+            } else {
+                bodyPretty += prettyjson.render(value, {}) + '\n';
             }
         });
 
-        output += format('[%s] %s\n%s\n',
-            time,
-            logLevel,
-            bodyPretty
-        );
-    }
-    // CASE: logging.info('text')
-    else if (data.msg) {
-        output += format('[%s] %s %s\n',
-            time,
-            logLevel,
-            data.msg
-        );
-    }
-    else {
-        output += format('[%s] %s\n',
-            time,
-            logLevel
-        );
+        if (this.mode !== 'short') {
```

Comment:
This is a TODO - we should probably move this if statement to prevent the calculation of bodyPretty if it is not going to be output.